pkg/controllers/termination/eviction.go

```diff
@@ -0,0 +1,87 @@
+package termination
+
+import (
+	"context"
+	"sync"
+
+	set "github.com/deckarep/golang-set"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/policy/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/util/workqueue"
+)
+
+type EvictionQueue struct {
+	queue        workqueue.RateLimitingInterface
+	coreV1Client corev1.CoreV1Interface
+
+	enqueued set.Set
+	once     sync.Once
+}
+
+// Evict adds a pod to the EvictionQueue
+func (e *EvictionQueue) Add(pods []*v1.Pod) {
```

Comment:
A variadic might be a bit cleaner here, since you could imagine evicting a single pod.

Reply:
Since the only code that is adding a pod is our code, I thought it would make more sense just to put an array here, since that's how we form the list of pods anyways. I have no strong feelings either way though.